src/rmp/src/Restructure.cpp

```diff
@@ -156,20 +161,25 @@ void Restructure::runABC()
 
   // abc optimization
   bool area_mode = opt_mode_ <= Mode::AREA_3;
-  std::string abc_script_file = work_dir_name_ + "ord_abc_script.tcl";
   std::string best_blif;
   int best_inst_count = std::numeric_limits<int>::max();
-  for (int opt_mode = static_cast<int>(area_mode)
+  int opt_mode_init = static_cast<int>(area_mode)
                           ? static_cast<int>(Mode::AREA_1)
                           : static_cast<int>(Mode::DELAY_1);
-       opt_mode <= (static_cast<int>(area_mode)
-                        ? static_cast<int>(Mode::AREA_3)
-                        : static_cast<int>(Mode::DELAY_4));
-       opt_mode++) {
-    opt_mode_ = (Mode) opt_mode;
+  int opt_mode_bound = static_cast<int>(area_mode)
+                           ? static_cast<int>(Mode::AREA_3)
+                           : static_cast<int>(Mode::DELAY_4);
+
+  std::vector<pid_t> child_proc(opt_mode_bound - opt_mode_init + 1, 0);
```

Comment:
@ahmadelrouby Need to keep this functionality optional with default off.

Reply:
Would also need to check on number of processes allowed. This would be user input as well. Default could be 4

Reply:
you should be getting that information from `Openroad::getThreadCount`